### PR TITLE
total sessions metric

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

**Comment by rudigiesler** [comment]
@hodgestar Ready for review.

File: lib/metrics-helper.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ var MetricsHelper = Eventable.extend(function(self, im) {
     self.add.total_unique_users = function(label) {
         /** MetricsHelper.add.total_unique_users(label)
 
-            Adds a sum metric that fires everytime a new session is started
+            Adds a sum metric that fires every time a new session is started
```
Comment (hodgestar) [inline lib/metrics-helper.js]: I realised belatedly that this should say "every time a new user is added"?

File: lib/metrics-helper.js [hunk 2]
```diff
@@ -31,6 +31,21 @@ var MetricsHelper = Eventable.extend(function(self, im) {
...
+        /** MetricsHelper.add.total_sessions(label)
+
+            Adds a sum metric that fires every time a new session is started.
+
+            :param string label:
+                The label for the metric. Defaults to ``total_sessions``
+        */
+        label = typeof label !== 'undefined' ? label: 'total_sessions';
+        self.im.on('session:new', function(e) {
+            return e.im.metrics.fire.sum(label, 1);
```
Comment (hodgestar) [inline lib/metrics-helper.js]: We should perhaps use `inc` here instead? `sum` doesn't produce a running total (the summing happens on 10s buckets only, and then someone has to sum all of the buckets later).

File: test/test_metrics_helper.js [hunk 3]
```diff
@@ -18,7 +18,9 @@ describe('MetricsHelper', function() {
             metricsH = new MetricsHelper(app.im);
             metricsH
                 .add.total_unique_users('uniqueUsers')
-                .add.total_unique_users();
+                .add.total_unique_users()
+                .add.total_sessions('sessions')
+                .add.total_sessions();
```
Comment (hodgestar) [inline test/test_metrics_helper.js]: Maybe we should split tests for each metric into different `.describe(...)` blocks and only to the setup relevant to the metric being tested in the `beforeEach` of the relevant `.describe(...)`? That would prevent this setup growing massively as new metrics are added.

Comment (hodgestar) [inline test/test_metrics_helper.js]: Ah, I see the tests are already split up so it's just a matter for moving the metric setup into the individual `.describe(...)` blocks.

**Comment by hodgestar** [comment]
Left a few comments. Otherwise looks good.

File: lib/metrics-helper.js (hunk 1 above)
Comment (rudigiesler) [inline lib/metrics-helper.js]: The full line is `every time a new session is started with a new user`

File: lib/metrics-helper.js (hunk 2 above)
Comment (rudigiesler) [inline lib/metrics-helper.js]: Should we not be using `inc` for the total unique users too then?

Also, is there a place where I can find the definitions of what each of the metric aggregation methods do?

File: test/test_metrics_helper.js (hunk 3 above)
Comment (rudigiesler) [inline test/test_metrics_helper.js]: Yes I'll move that into each `describe` block

File: lib/metrics-helper.js (hunk 1 above)
Comment (hodgestar) [inline lib/metrics-helper.js]: Ah. Woot. :)

File: lib/metrics-helper.js (hunk 2 above)
Comment (hodgestar) [inline lib/metrics-helper.js]: Yep, we should probably be using `inc` for total unique users too (sorry for not spotting that earlier).

All the metric aggregation functions happen over time bucket intervals (configurable, but 10s by default). You can find all of the aggregation function implementations at https://github.com/praekelt/vumi/blob/develop/vumi/blinkenlights/metrics.py#L201-L206.

You'll notice that `inc` isn't one of the aggregate types. `inc` is implemented manually using a counter in Redis and fires a `last` metric with the most recent value. See https://github.com/praekelt/vumi-jssandbox-toolkit/blob/master/lib/metrics/api.js#L103.

**Comment by hodgestar** [comment]
:+1:

**Comment by rudigiesler** [comment]
@hodgestar changed `total_unique_users` to use the `inc` metric aggregator. Can you check if everything is still okay?

**Comment by hodgestar** [comment]
@rudigiesler: Still looks good. Second :+1:.

**Pull request merged**